docs/en/engines/table-engines/special/distributed.md

```diff
@@ -31,6 +31,12 @@ Also it accept the following settings:
 
 - `fsync_directories` - do the `fsync` for directories. Guarantees that the OS refreshed directory metadata after operations related to asynchronous inserts on Distributed table (after insert, after sending the data to shard, etc).
 
+- `bytes_to_throw_insert` - if more than this number of compressed bytes will be pending for async INSERT, an exception will be thrown. 0 - do not throw. Default 0.
```

Comment:
1) Actually number of files collected can also be important (1 mln of tiny files in same folder is a problem, even if their overall size is small).

2) Maybe we should set up some reasonable big default? Like 10-20 GB?

Reply:
>Actually number of files collected can also be important (1 mln of tiny files in same folder is a problem, even if their overall size is small).

Right now number of files can be limited by using `distributed_directory_monitor_batch_inserts`, although it is not enabled by default (and with default squashing it will create pretty small files anyway), so I'm agree, limiting based on number of files may also be a good idea.

Since This PR already contains some unrelated patches and plus it was hanged, so this will be done separately.

>Maybe we should set up some reasonable big default? Like 10-20 GB?

I though about this, but decided not to change the defaults (for example I have one use case when I don't need to limit it), since the client that INSERT into Distributed may not be ready for this type of errors.